### PR TITLE
log message while events are still draining

### PR DESCRIPTION
**Pull request by jakekaplan**
Log a message while items are still draining out of the `QueueService`. This is particularly important for UX of client side task run orchestration. Consider this example:

```python
from prefect import task, flow

@task
def add_one(n):
    return n + 1


@flow
def add_flow():
    futures = [add_one.submit(i) for i in range(1000)]
    for future in futures:
        future.result()


if __name__ == '__main__':
    add_flow()
```

1000 tasks will all finish at roughly the same time and generate 1000 events. The flow will finish and it will look like the process is hanging to the user. In reality we still need to churn through those 1000 events. This PR adds the following logging when the service begins to drain.

```
$ python add_flow.py
11:29:22.461 | INFO    | prefect.engine - Created flow run 'celadon-scorpion' for flow 'add-flow'
...
... 
...
11:30:29.164 | INFO    | Task run 'add_one-8d6' - Finished in state Completed()
11:30:29.165 | INFO    | Task run 'add_one-767' - Finished in state Completed()
11:30:29.165 | INFO    | Task run 'add_one-c4a' - Finished in state Completed()
11:30:29.166 | INFO    | Task run 'add_one-c10' - Finished in state Completed()
11:30:29.629 | INFO    | Flow run 'celadon-scorpion' - Finished in state Completed()
11:30:29.637 | WARNING | GlobalEventLoopThread | prefect._internal.concurrency - EventsWorker is still processing: 941 items remaining...
11:30:32.644 | WARNING | GlobalEventLoopThread | prefect._internal.concurrency - EventsWorker is still processing: 717 items remaining...
11:30:35.651 | WARNING | GlobalEventLoopThread | prefect._internal.concurrency - EventsWorker is still processing: 489 items remaining...
11:30:38.656 | WARNING | GlobalEventLoopThread | prefect._internal.concurrency - EventsWorker is still processing: 258 items remaining...
11:30:41.664 | WARNING | GlobalEventLoopThread | prefect._internal.concurrency - EventsWorker is still processing: 30 items remaining...
```

Looking for feedback on both the approach and the actual copy/UX!

### CODE REVIEW

File: src/prefect/settings.py [hunk 1]
```diff
@@ -828,12 +828,12 @@ def default_cloud_ui_url(settings, value):
 
 PREFECT_LOGGING_INTERNAL_LEVEL = Setting(
     str,
-    default="ERROR",
+    default="WARNING",
```
Comment (jakekaplan) [inline src/prefect/settings.py:831]: I feel like it is okay to move this to `WARNING` so we can emit things we care about from the service. Alternatively could write to `sys.stderr` directly? Or can investigate taking a totally different approach to this pr.

Comment (desertaxle) [reply src/prefect/settings.py:831]: I'd rather keep this at `ERROR` and use a different logger for the new message. The `GlobalEventLoopThread | prefect._internal.concurrency` in the logs you posted in the PR description seems like unnecessary information.

Comment (jakekaplan) [reply src/prefect/settings.py:831]: that's fair! let me see what I can do

Comment (chrisguidry) [reply src/prefect/settings.py:831]: +1 exposing that `_internal` package feels icky

Comment (jakekaplan) [reply src/prefect/settings.py]: Okay I've updated the service to have it's own logger. I am somewhat scared about this just based on this file: https://github.com/PrefectHQ/prefect/blob/d2de43322241d1e5668684b6047e76ee4ebd7830/src/prefect/_internal/_logging.py#L1-L26

but now it looks like this:
```
13:36:30.706 | INFO    | Task run 'add_one-b72' - Finished in state Completed()
13:36:30.706 | INFO    | Task run 'add_one-df7' - Finished in state Completed()
13:36:31.154 | INFO    | Flow run 'ninja-weasel' - Finished in state Completed()
13:36:31.156 | WARNING | EventsWorker - Still processing items: 941 items remaining...
13:36:34.159 | WARNING | EventsWorker - Still processing items: 715 items remaining...
13:36:37.163 | WARNING | EventsWorker - Still processing items: 482 items remaining...
13:36:40.174 | WARNING | EventsWorker - Still processing items: 248 items remaining...
13:36:43.181 | WARNING | EventsWorker - Still processing items: 18 items remaining...
```

Comment (jakekaplan) [reply src/prefect/settings.py]: I could add another implementation of `SafeLogger` that doesn't depend on `PREFECT_LOGGING_INTERNAL_LEVEL`?

**Comment by jakekaplan** [comment]
going to re-open this pr against `main`

File: src/prefect/settings.py (hunk 1 above)
Comment (jakekaplan) [reply src/prefect/settings.py]: re-opened here: https://github.com/PrefectHQ/prefect/pull/14964

**Pull request closed without merging**